### PR TITLE
Arf 80526 Create arp-dev-template.ejs

### PR DESCRIPTION
**Pull request by williamphelps13**
## Summary

- Files changed
  - config/webpack.config.js
  - src/platform/landing-pages/arp-dev-template.ejs
- What is the solution? 
  - Before this change, the development template was hardcoded to dev-template.ejs which is meant to simulate the content-build page-react.html page while in development
  - Looking at the content-build registry.json all applications that have a layout use page-react.html as the layout except the accredited-representative-portal application which uses the accredited-representative-portal.html layout
  - Because the accredited-representative-portal application does not use page-react.html it should also not use dev-template.ejs and should instead have a dev template that simulates accredited-representative-portal.html in development
  - This PR adds arp-dev-template.ejs that simulates accredited-representative-portal.html in development
  - To make arp-dev-template.ejs be the template for the accredited-representative-portal application in development this PR also updates config/webpack.config.js making all applications first check to see if they use a layout that is in a
layout map (this will only be true for the accredited-representative-portal application) and if they do not then it falls back to what was previously the only option dev-template.ejs
- Why is this the solution?
  - This change Includes code that all applications will run in development, but only the accredited-representative-portal application will actually have code that evaluates differently. All other applications will continue to use the dev-template.ejs. 
  - This implementation allows other applications to use the arp-dev-template.ejs or add their own dev template by just adding a key and value to the map
  - This change is necessary because having dev-template.ejs surround the accredited-representative-portal application made dev and staging environments contrast sharply for the accredited-representative-portal application. Not only is this visually distracting for demos, but there were multiple instances of failing e2e tests in dev that would not be failing in staging or prod
- Which team do you work for, does your team own the maintenance of this component?
  - Accredited Representative Facing Team
  - The ARF team will own maintenance of src/platform/landing-pages/arp-dev-template.ejs
  - The ARF team will not own maintenance of config/webpack.config.js
 

## Related issue(s)

- [Remove vets-website dev-template.ejs as ARF development layout #80526](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/80526)
- [[ARF] Implement Header and Footer#78055](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/78055)

## Testing done

- When in development, developers navigated to /representative they would see the dev-template.ejs surrounding the accredited-representative-portal which since dev-template.ejs is meant to simulate page-react.html was not simulating staging accurately
- When in development, developers navigate to /representative they see the arp-dev-template.ejs surrounding the accredited-representative-portal which since arp-dev-template.ejs is meant to simulate accredited-representative-portal.html simulates staging more accurately
- There don't appear to be any tests for dev-template.ejs because it is just used in dev so there are also no task included for arp-dev-template.ejs
#### Before
Screenshot 2024-05-24 at 1 57 38 PM
Screenshot 2024-05-24 at 1 57 24 PM

#### After
Screenshot 2024-05-24 at 2 04 31 PM
Screenshot 2024-05-24 at 2 04 52 PM

#### Before
Screenshot 2024-05-24 at 1 49 14 PM
Screenshot 2024-05-24 at 1 47 17 PM

#### After
Screenshot 2024-05-24 at 1 43 45 PM
Screenshot 2024-05-24 at 1 44 14 PM

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- NA I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

**Review by gabezurita — COMMENTED** [review]
Looking good so far! I'll give it a review once it's out of draft!

File: config/webpack.config.js [hunk 1]
```diff
@@ -210,7 +215,9 @@ function generateHtmlFiles(buildPath, scaffoldAssets) {
       filename: path.join(buildPath, rootUrl, 'index.html'),
       inject: false,
       scriptLoading: 'defer',
-      template: 'src/platform/landing-pages/dev-template.ejs',
+      template:
+        templateLayoutToDevTemplate[template.layout] ||
+        'src/platform/landing-pages/dev-template.ejs',
```
Comment (gabezurita) [inline config/webpack.config.js:220]: The inline conditional is straightforward but might benefit from a helper function for readability and potential reuse.

```javascript
function getDevTemplate(templateLayout) {
  return templateLayoutToDevTemplate[templateLayout] || 'src/platform/landing-pages/dev-template.ejs';
}
...
template: getDevTemplate(template.layout),
...
```

**Review by gabezurita — DISMISSED** [review]
LGTM! One idea:

**Review by gabezurita — APPROVED** [review]
Nice improvement on the last commit! Still LGTM!

**Review by ataker — APPROVED** [review]

**Pull request merged**